src/clr-angular/data/datagrid/datagrid.spec.ts

```diff
@@ -211,16 +213,89 @@ export default function(): void {
             });
         });
 
+        describe("Content Projection", function() {
+            let context: TestContext<ClrDatagrid, FullTest>;
+            let displayMode: DisplayModeService;
+            describe("when displaying", function() {
+                let displayHeader: Element;
+                let displayColumns: NodeList;
+                let displayTable: Element;
+                let displayRows: NodeList;
+
+                beforeEach(function() {
+                    context = this.create(ClrDatagrid, FullTest, [HideableColumnService]);
+                    displayMode = context.getClarityProvider(DisplayModeService);
+                });
+
+                it("moves columns into the display container", function() {
+                    context.clarityDirective.displayMode.view.subscribe(mode => {
+                        displayHeader = context.clarityElement.querySelector(".datagrid-header");
+                        displayColumns = displayHeader.querySelectorAll(".datagrid-column");
+                        if (mode === DatagridDisplayMode.DISPLAY) {
+                            expect(displayColumns.length).toBe(2);
```

Comment:
Why is all of this in the subscription? You can just inline it after `displayMode.updateView(DatagridDisplayMode.DISPLAY);` and it's much safer. As usual, tests that put `expect` statements in the subscription will always pass if the Observable simply doesn't fire.